Source/Private/ASDisplayNode+UIViewBridge.mm

```diff
@@ -796,15 +796,44 @@ - (void)setBackgroundColor:(UIColor *)newBackgroundColor
 - (UIColor *)tintColor
 {
   _bridge_prologue_read;
-  ASDisplayNodeAssert(!_flags.layerBacked, @"Danger: this property is undefined on layer-backed nodes.");
-  return _getFromViewOnly(tintColor);
+  if (_loaded(self)) {
+    if (_flags.layerBacked) {
+        // The first nondefault tint color value in the view’s hierarchy, ascending from and starting with the view itself.
+        return _tintColor ?: self.supernode.tintColor;
```

Comment:
Our locking principles don't allow walking up the hierarchy while holding the node's lock. So you'd need to do it after releasing the lock.